src/aleph_client/vm/cache.py

```diff
@@ -1,56 +1,121 @@
         if not re.match(r"^\w+$", key):
             raise ValueError("Key may only contain letters, numbers and underscore")
         data = value if isinstance(value, bytes) else value.encode()
         async with self.session.put(
-            f"{settings.API_HOST}/cache/{key}", data=data
+            f"{self.api_host}/cache/{key}", data=data
         ) as resp:
             resp.raise_for_status()
             return await resp.json()
 
-    async def delete(self, key: str):
+    async def delete(self, key: str) -> Any:
         if not re.match(r"^\w+$", key):
             raise ValueError("Key may only contain letters, numbers and underscore")
-        async with self.session.delete(f"{settings.API_HOST}/cache/{key}") as resp:
+        async with self.session.delete(f"{self.api_host}/cache/{key}") as resp:
             resp.raise_for_status()
             return await resp.json()
 
-    async def keys(self, pattern: str = "*"):
+    async def keys(self, pattern: str = "*") -> List[str]:
         if not re.match(r"^[\w?*^\-]+$", pattern):
             raise ValueError(
-                "Patterh may only contain letters, numbers, underscore, ?, *, ^, -"
+                "Pattern may only contain letters, numbers, underscore, ?, *, ^, -"
             )
         async with self.session.get(
-            f"{settings.API_HOST}/cache/?pattern={pattern}"
+            f"{self.api_host}/cache/?pattern={pattern}"
         ) as resp:
             resp.raise_for_status()
             return await resp.json()
+
+
+class TestVmCache(BaseVmCache):
+    """This is a local, dict-based cache that can be used for testing purposes."""
+
+    def __init__(self, session: Optional[ClientSession] = None):
+        super().__init__(session)
+        self._cache: Dict[str, bytes] = {}
+
+    async def get(self, key: str) -> Optional[bytes]:
+        if not re.match(r"^\w+$", key):
+            raise ValueError("Key may only contain letters, numbers and underscore")
+        return self._cache.get(key)
+
+    async def set(self, key: str, value: Union[str, bytes]) -> None:
+        if not re.match(r"^\w+$", key):
+            raise ValueError("Key may only contain letters, numbers and underscore")
+        data = value if isinstance(value, bytes) else value.encode()
+        self._cache[key] = data
+
+    async def delete(self, key: str) -> None:
+        if not re.match(r"^\w+$", key):
+            raise ValueError("Key may only contain letters, numbers and underscore")
+        del self._cache[key]
+
+    async def keys(self, pattern: str = "*") -> List[str]:
+        if not re.match(r"^[\w?*^\-]+$", pattern):
+            raise ValueError(
+                "Pattern may only contain letters, numbers, underscore, ?, *, ^, -"
+            )
+        all_keys = list(self._cache.keys())
+        if pattern == "*":
+            return all_keys
+        return [key for key in all_keys if re.match(pattern, key)]
```

Comment:
Seems like we're mixing both regular expressions and glob expressions here. What's the actual spec?

Reply:
Keys may only be composed of letters and digits that `re.match(r"^\w+$", key)`.

Reply:
Yes, what bothers me here is the pattern. I see `if pattern == "*":` and then `re.match(pattern, key)` so there's something to clarify.

Reply:
Whoops, you're right! These are wildcard and not regular expressions: https://redis.io/docs/stack/search/reference/query_syntax/